### ISSUE TITLE
Throw ApplicationError in lifecycles.ts doesn't show error in admin panel

### ISSUE THREAD

**Issue by ngaritagoitia** [strapi/strapi#20343]
### Required System information

Environment: development
OS: linux-x64
Strapi Version: 4.24.2
Node/Yarn Version: npm/10.5.0 node/v20.12.2 linux x64 workspaces/false
Edition: Community
Database: mysql2

### Describe the bug

throwing ApplicationError from strapi utils doesn't show an error in the admin ui. I get a 400 in the terminal, showing that the error was thrown, but in previous versions of strapi, an error would appear at the top of the admin panel with the custom error message
### Steps to reproduce the behavior

add this code to a lifecycles.ts and see how no error is shown in the admin panel:

```ts
export default {
  beforeCreate(event) {
    throw new ApplicationError('Test', 400);

  }
}
```

### Expected behavior

An error to be shown at the top of the admin panel with the custom error.

**Comment by ngaritagoitia** [comment]
Yeah it's something to do with axios, we had to manually specify `"axios": "1.6.0"` in the package.json of our monorepo

**Comment by ngaritagoitia** [comment]
https://github.com/strapi/strapi/issues/19495#issuecomment-2220058205

**Comment by alexandermirzoyan** [comment]
We don't have manually installed `axios` package in our app, I look inside `node_modules` folder and `axios` is `v1.6.0` but still we have the problem

**Comment by derrickmehaffy** [comment]
> We don't have manually installed `axios` package in our app, I look inside `node_modules` folder and `axios` is `v1.6.0` but still we have the problem

Check any plugins you installed as well

**Comment by alexandermirzoyan** [comment]
Just figured out that my problem is when throwing error from global middleware which works for `/upload/actions/bulk-delete` endpoint and DELETE `/upload/files/` so I want to throw message that the file is in use.

**Comment by derrickmehaffy** [comment]
We recently upgraded our internal usage of axios, can you all try upgrading to today's release and see if the issue still happens?

**Comment by alexandermirzoyan** [comment]
Tried to put `v4.25.9` but still the issue is not fixed.

**Comment by ArtyomSukiasyan** [comment]
If you want to throw error in the Strapi API, writing such a middleware solves the problem, but it will not work if you want to throw an error in some plugin (like upload plugin).

1. Create in strapi directory **/middlewares/errorHandler.index.js** and add code below in the file.

```javascript
module.exports = () => {
  return async (ctx, next) => {
    try {
      await next();
    } catch (err) {
      const statusCode = getStatusCodeByErrorName(err.name);
      ctx.status = statusCode;
      ctx.body = {
        data: null,
        error: { ...err, status: statusCode },
      };
      if (ctx.app.emit) {
        ctx.app.emit("error", err, ctx);
      }
    }
  };
};

function getStatusCodeByErrorName(name) {
  switch (name) {
    case "ValidationError":
      return 400;
    case "AuthenticationError":
      return 401;
    case "ForbiddenError":
      return 403;
    case "NotFoundError":
      return 404;
    default:
      return 500;
  }
}
```

2. Go to your Strapi main directory **/config/middlewares.js** file and add this code:
```javascript
  {
    resolve: "./middlewares/errorHandler",
    config: {
      enabled: true,
    },
  },

**Comment by alexandermirzoyan** [comment]
Agree @ArtyomSukiasyan, this is a nice solution but it does not work when you write some global middleware for Media Library plugin.

**Comment by 00sapo** [comment]
Seems a problem with axios: https://github.com/strapi/strapi/issues/19495

**Comment by cdreier** [comment]
Hi, we just ran into the same issue, throwing an ApplicationError from a plugin, versions are in sync, axios version pinning tested but had no effect in this case - debugged to this point, were the error is just no (or no longer?) an instance of ApplicationError

![image](https://github.com/user-attachments/assets/81510c49-ed62-4502-8d83-7ff667914606)

**Comment by cdreier** [comment]
one more addition, it seems like the transpiled plugin code matches no longer the "core" util ApplicationErrors - even if constructors etc are there and looks the same.

as a quickfix, i adapted the idea from @ArtyomSukiasyan with a middleware that catches the error, simply checks the name and throws again the correct error which then shows also up on the ui

```ts
import { errors } from "@strapi/utils"

export default () => {
  return async (ctx, next) => {
    try {
      await next();
    } catch (err) {
      if (err.name === 'ApplicationError') {
        throw new errors.ApplicationError(err.message, err.details);
      }
      throw err
    }
  };
};
```
